EpistasisLab/tpot2
# Initial sklearnex support

[please review the [Contribution Guidelines](http://epistasislab.github.io/tpot/contributing/) prior to submitting your pull request. go ahead and delete this line if you've already reviewed said guidelines.]

## What does this PR do?

Adds configs for and enables use of the Intel(R) Extension for Scikit-learn in tpot2.

## How should this PR be tested?

Ensure backwards compatibility - make sure previous configurations run without issue.
Test usage of the config files both using TPOTEstimator and TPOTRegressor/TPOTClassifier with root_config_dict set to classifiers_sklearnex or regressors_sklearnex respectively.

## Any background context you want to provide?

https://github.com/intel/scikit-learn-intelex/
Notebook with demonstration of speedup capabilities in tpot: https://colab.research.google.com/drive/1yeqbKg2Sv8affoHrOCsluunscjQnmqy5#scrollTo=h_TKDu-FnAFZ&line=5&uniqifier=1

## What are the relevant issues?

[#1316](https://github.com/EpistasisLab/tpot/issues/1316)

## Questions:

- Should a requirements-optional.txt be created?
- I would like to add an example, I see notebooks are primarily being used here which would be nice to demonstrate opportunities for speedups - any reservations about this?
